python/ray/data/_internal/plan.py

```diff
@@ -692,6 +687,27 @@ def has_computed_output(self) -> bool:
             and not self._snapshot_blocks.is_cleared()
         )
 
+    def _run_with_new_execution_backend(self) -> bool:
+        """Whether this plan should run with new backend."""
+        from ray.data._internal.stage_impl import RandomizeBlocksStage
+
+        # Read-equivalent stage is handled with the legacy execution impl for now,
+        # except it's read->randomize_block_order stage.
```

Comment:
This could use some clarification.

```
# Read only: handle with legacy backend
# Read->randomize_block_order: handle with new backend
# Note that both are considered read equivalent, hence this extra check.
```

Reply:
Done.